lib/Cake/Cache/CacheEngine.php

```diff
@@ -84,6 +84,16 @@ public function gc($expires = null) {
 	abstract public function write($key, $value, $duration);
 
 /**
+ * Write value for a key into cache if it doesn't already exist
+ *
+ * @param string $key Identifier for the data
+ * @param mixed $value Data to be cached
+ * @param int $duration How long to cache for.
+ * @return bool True if the data was successfully cached, false on failure
+ */
+	abstract public function add($key, $value, $duration);
```

Comment:
Instead of an abstract method, this should probably be an empty method. Right now this will be a breaking change as people upgrade. We try to keep upgrades as painless as possible in the minor releases.